### PR TITLE
Better filesystem handling in Settings.cs

### LINKED ISSUE #836: Should Settings.xml be checked into the repository?
I can see something like `DefaultSettings.xml` being checked in, but not the file that is actually modified for saving and loading settings. This means that ` git checkout -- Assets/StreamingAssets/Settings/Settings.xml` is necessary every time someone wants to switch branches after making settings changes for testing things. Or git stash and apply, but that's still hassle.

So could we have a `DefaultSettings.xml` which is checked into the repo and loaded first by the game, then a `Settings.xml` or `UserSettings.xml` which is overridden by that and not tracked by git?

### PR DESCRIPTION
**Pull request by mikejbrown**
Fixes #836 

Adds existence checks for settings file, first looking in the user's persistent data folder, then the StreamingAssets Settings.xml and then falls back to hard coded defaults if there are problems.

Also some minor StyleCop fixes.

### CODE REVIEW

File: Assets/Scripts/Utilities/Settings.cs [hunk 1]
```diff
@@ -111,26 +135,53 @@ private static void saveSettings()
...
         settingsDict = new Dictionary<string, string>();
+        string furnitureXmlText;
 
-        // get file path of Settings.xml and load the text
-        string filePath = System.IO.Path.Combine(Application.streamingAssetsPath, settingsFilePath);
-        string furnitureXmlText = System.IO.File.ReadAllText(filePath);
+        // load the settings XML file
+        // first try the user's private settings file in userSettingsFilePath
+        // if that doesn't work fall back to defaultSettingsFilePath
+        // if that doesn't work fall back to the hard coded furnitureXmlText above
```
Comment (Randshot) [inline Assets/Scripts/Utilities/Settings.cs]: Comments should start with a capital letter and also end with a period.

**Comment by Randshot** [comment]
Works for me without any errors and the code changes look clean too (except for the above mentioned comments).

**Comment by mikejbrown** [comment]
Made style improvements :smile:

File: Assets/Scripts/Utilities/Settings.cs [hunk 2]
```diff
@@ -149,15 +152,15 @@ private static void saveSettings()
 
     private static void loadSettings()
     {
-        // initilize the settings dict
+        // Initilize the settings dict.
```
Comment (mikejbrown) [inline Assets/Scripts/Utilities/Settings.cs]: *&^#ing spelling mistake. Of course I catch it immediately _after_ doing the push. :stuck_out_tongue_closed_eyes:

File: Assets/Scripts/Utilities/Settings.cs [hunk 3]
```diff
@@ -178,23 +181,23 @@ private static void loadSettings()
...
         Debug.Log(xDoc.InnerText); // Uber Logger doesn't handle multilines.
 
-        // get the Settings node , it's children are the individual settings
+        // Get the Settings node. Its children are the individual settings.
         XmlNode settingsNode = xDoc.GetElementsByTagName("Settings").Item(0);
         XmlNodeList settingNodes = settingsNode.ChildNodes;
         Debug.ULogChannel("Settings", settingNodes.Count + " settings loaded");
 
-        // loop for each setting
+        // Loop for each setting
```
Comment (Randshot) [inline Assets/Scripts/Utilities/Settings.cs]: I think you forgot one period 👅

**Comment by Randshot** [comment]
@mikejbrown You generally don't have to fix styling issues for code you didn't change or wrote yourself.

File: Assets/Scripts/Utilities/Settings.cs (hunk 3 above)
Comment (mikejbrown) [inline Assets/Scripts/Utilities/Settings.cs]: Oh for fshks sake! :confounded: I'm stopping here. :stuck_out_tongue:

**Comment by mikejbrown** [comment]
@Randshot I know. I'm just that kind of guy. :sigh:

**Comment by mikejbrown** [comment]
If I don't do enough commits to force git into a SHA-1 collision I feel like I'm letting the team down. :wink:

**Comment by Randshot** [comment]
Well, I think you have some work ahead of you :)

**Comment by TomMalbran** [comment]
I kind of don't like having the default xml code pasted inside the code. It feels weird. I would prefer to have it another file or maybe in code. But that will depend on what will be the best option for modding. Creating default settings through code or with an xml file.

**Comment by mikejbrown** [comment]
@TomMalbran The "real" defaults are in the StreamingAssets Settings.xml file. The hard coded values are there if the defaults file gets borked by a mod or something. And they could always be overridden by code.

**Comment by TomMalbran** [comment]
Mods will not write over the default file, each mod will have its own settings file in its folder. The only ones that can change it are the users, and if they break it, is their fault? same as if they break the furniture stuff. If the file can't be read or something like we can just throw a big error.

We aren't doing anything with the other files in streaming assets that the users can change, so I am not sure if we need it here. We can discuss that in another topic if required.

**Comment by Mizipzor** [comment]
This branch has conflicts that must be resolved, probably due to the huge StyleCop cleanup that just happened. Fix them and remove the `[Conflicts]` tag. :)

**Comment by mikejbrown** [comment]
Fixed merge conflicts

File: Assets/Scripts/Utilities/Settings.cs [hunk 4]
```diff
@@ -153,65 +179,124 @@ private static string GetSetting(string key)
...
-        // Get file path of Settings.xml and load the text.
-        string filePath = System.IO.Path.Combine(Application.streamingAssetsPath, settingsFilePath);
-        string furnitureXmlText = System.IO.File.ReadAllText(filePath);
+        // Load the settings XML file.
+        // First try the user's private settings file in userSettingsFilePath.
+        // If that doesn't work fall back to defaultSettingsFilePath.
+        // If that doesn't work fall back to the hard coded furnitureXmlText above.
+        if (System.IO.File.Exists(userSettingsFilePath) == false)
+        {
+            Debug.ULogWarningChannel("Settings", "User settings file could not be found at '" + userSettingsFilePath + "'. Falling back to defaults.");
```
Comment (Mizipzor) [inline Assets/Scripts/Utilities/Settings.cs]: This cant be a warning, its expected to not exist the first time you start the game. Log it if you want, but not as a problem. :)

**Comment by mikejbrown** [comment]
@Mizipzor done :smile:

**Comment by mikejbrown** [comment]
Sweet, thanks :cake:

**Pull request merged**